### DISCUSSION TITLE
New AIO installation fails to start database

### DISCUSSION THREAD

**Discussion by aradian** [nextcloud/all-in-one#5396] [Questions] [answered]
### Steps to reproduce
1. Run docker command (see below), starting with no containers or volumes
2. Log in to AIO console at hostname:8080
3. Click the "Start and update containers" button with default optional containers selected

### Expected behavior 
All containers start.

### Actual behavior 
Redis, Collabora, Talk, Imaginary show as green.
Apache, Notify Push, Nextcloud, Database show as yellow

Nextcloud logs are repeating:
nc: getaddrinfo for host "nextcloud-aio-database" port 5432: Try again
2024-10-05T01:47:43.392946400Z Waiting for database to start...
2024-10-05T01:47:53.401290999Z nc: getaddrinfo for host "nextcloud-aio-database" port 5432: Try again
2024-10-05T01:47:53.402069361Z Waiting for database to start...

Database logs are repeating:
Database export failed the last time. Most likely was the export time not high enough.
2024-10-05T01:50:03.603497938Z Please report this to https://github.com/nextcloud/all-in-one/issues. Thanks!

#### Host OS 
NAME="Linux Mint"
VERSION="21.1 (Vera)"

#### Output of `sudo docker info`
Client: Docker Engine - Community
 Version:    24.0.2
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.10.5
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.18.1
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 10
  Running: 1
  Paused: 0
  Stopped: 9
 Images: 14
 Server Version: 24.0.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: false
  userxattr: true
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 5.15.0-117-generic
 Operating System: Linux Mint 21.1
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 15.44GiB
 Name: nextcloud
 ID: d0318484-fa75-44b6-910e-636b60efdfee
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

#### Docker run command or docker-compose file that you used
sudo docker run \
        --init \
        --sig-proxy=false \
        --name nextcloud-aio-mastercontainer \
        --restart always \
        --publish 80:80 \
        --publish 8080:8080 \
        --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
        --volume /var/run/docker.sock:/var/run/docker.sock:ro \
        --env NEXTCLOUD_DATADIR="/media/userdata/nextcloud_datadir" \
        --env SKIP_DOMAIN_VALIDATION=true \
        nextcloud/all-in-one:latest

#### Other valuable info 
Trying to create a backup fails quickly, with log message:
The .ncdata or .ocdata file is missing in Nextcloud datadir which means it is invalid!
2024-10-05T02:17:38.526225906Z Is the drive where the datadir is located on still mounted?

/media/userdata/nextcloud_datadir has an empty file created: "skip.update"
/media/userdata/nextcloud-aio-backup empty directory is created

**Comment by szaimen** [comment]
Hi, I think the reason is that docker is not up-to-date. (e.g. v27.1) Can you upgrade if with `sudo apt update; sudo apt upgrade -y` ?

**Comment by szaimen** [comment]
Duplicate of https://github.com/nextcloud/all-in-one/discussions/4390

**Comment by aradian** [comment]
After upgrading all packages and rebooting, I'm seeing the same result as previously.

"Database export failed the last time. Most likely was the export time not high enough."
Regarding this message, is it possible to increase the allowed export time? Or skip the database export on startup? I wasn't able to find info about this in the docs.

Thanks

New docker info:
Client: Docker Engine - Community
 Version:    27.3.1
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.17.1
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.7
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 10
  Running: 1
  Paused: 0
  Stopped: 9
 Images: 13
 Server Version: 27.3.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: true
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7f7fdf5fed64eb6a7caf99b3e12efcf9d60e311c
 runc version: v1.1.14-0-g2c9f560
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 5.15.0-117-generic
 Operating System: Linux Mint 21.1
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 15.44GiB
 Name: nextcloud
 ID: d0318484-fa75-44b6-910e-636b60efdfee
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

**Comment by szaimen** [comment]
Did you read one comment up? https://github.com/nextcloud/all-in-one/discussions/4390#discussioncomment-10873342

**Comment by aradian** [comment]
Yes. I think that is a different issue than what I'm seeing.

- I'm getting a different error related to the database and am not trying to run a backup
- Mint 21.1 is based on Ubuntu 22.04
- fuse is already the newest version (2.9.9-5ubuntu3)

The machine running Nextcloud has a fairly slow hard drive and the error message seems to be implying some kind of timeout is occurring.

**Comment by szaimen** [comment]
Ah I see. Is it this error?
https://github.com/nextcloud/all-in-one/blob/531104db66b9b0467b6620caa9b8929600ef9d02/Containers/postgresql/start.sh#L59-L63

**Comment by szaimen** [comment]
What was the former AIO version from which you updated?

**Comment by aradian** [comment]
Yes, that's the error. I'm not sure what the old version was. I tried to wipe it out to start a new installation by killing all docker containers and then "docker container prune", "docker volume prune", and deleting everything in the nextcloud datadir.

**Comment by szaimen** [comment]
> Yes, that's the error. I'm not sure what the old version was. I tried to wipe it out to start a new installation by killing all docker containers and then "docker container prune", "docker volume prune", and deleting everything in the nextcloud datadir.

I see. That should make it work if you dont need to keep the data. Did you follow https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-properly-reset-the-instance?

**Comment by aradian** [comment, marked as answer]
Thanks for the link, I hadn't seen that. After docker volume prune there are a number of volumes remaining:

docker volume ls --filter "dangling=true"
DRIVER    VOLUME NAME
local     nextcloud_aio_apache
local     nextcloud_aio_backup_cache
local     nextcloud_aio_database
local     nextcloud_aio_database_dump
local     nextcloud_aio_elasticsearch
local     nextcloud_aio_mastercontainer
local     nextcloud_aio_nextcloud
local     nextcloud_aio_redis

After deleting those, the new AIO instance starts up.
